table of content for in-app updates (#594)
appcircleio/appcircle-docusaurus: docs/enterprise-appstore/in-app-updates/index.md

---
title: In-app Updates
description: Streamline app maintenance and enhance user experience by implementing in-app updates through your Enterprise App Store, ensuring seamless and automatic version upgrades.
tags:
  [
    enterprise app store,
    app distribution,
    app deployment,
    enterprise apps,
    app store setup,
    appcircle app store,
    In-app updates,
  ]
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import Screenshot from '@site/src/components/Screenshot';

In-app updates enable applications to deliver and install updates directly within the app, enhancing user experience by minimizing disruption.

##   What are In-app Updates for Streamlined Update Experience

> In-app updates provide a seamless way to deliver and install new versions of an application directly from within the app, eliminating the need for users to manually check for updates. This streamlined approach ensures users receive the latest features and fixes efficiently, enhancing their overall experience and keeping the app up-to-date.

Reviewer: ```suggestion
In-app updates offer a seamless method for delivering and installing new versions of an application directly within the app. This eliminates the need for users to manually check for updates, ensuring they receive the latest features and fixes efficiently. This streamlined approach enhances the overall user experience and keeps the app up-to-date.
```

##  Benefits and Examples of In-app Updates

> In-app updates offer several benefits, including a smoother user experience by allowing seamless updates without requiring users to manually download or install new versions. Examples include automatically applying critical bug fixes and feature enhancements while the app is running, thereby ensuring users always have access to the latest improvements and functionalities.

Reviewer: ```suggestion
In-app updates offer several benefits, including a smoother user experience by enabling seamless updates without requiring users to manually download or install new versions. For example, critical bug fixes and feature enhancements can be automatically applied while the app is running, ensuring users always have access to the latest improvements and functionalities.
```

## Implementing In-App Updates

##  Prerequisites for Integration

###  Authentication Requirements

To integrate an in-app update experience, you will need the **Personal Access Token (PAT)**, the **enterprise store prefix**, and the **enterprise store profile ID**.

### How to Obtain Integrations Parameters

#### Personal Access Token (PAT)

> - For details on generating an Appcircle Personal Access Token, visit [Generating/Managing Personal API Tokens](/appcircle-api/api-authentication#generatingmanaging-the-personal-api-tokens).

Reviewer: ```suggestion
- For details on generating an Appcircle Personal Access Token, please visit the [Generating/Managing Personal API Tokens](/appcircle-api/api-authentication#generatingmanaging-the-personal-api-tokens).
```

#### Enterprise Store Prefix

Navigate to the Enterprise Store module and settings page to find the **STORE PREFIX** information. You can also modify it if needed.

<Screenshot url='https://cdn.appcircle.io/docs/assets/BE_4207-Enterprise-Store-Prefix-1.png' />

#### Enterprise Store Profile Id

You can obtain your Enterprise Store Profile ID from the URL or by using the @appcircle/cli.

##### How to Extract Your Enterprise Store Profile ID from the URL

1. Navigate to your Enterprise Store Profile.
2. Check the URL, which should be in this format: **/enterprise-store/profiles/PROFILE_ID**. The PROFILE_ID refers to your specific profile ID.

##### Retrieving Profile ID Using @appcircle/cli

The upcoming command retrieves the complete list of Enterprise Store Profiles.

```bash
appcircle enterprise-app-store profile list
```

### Authentication for Updates

#### Retrieving Access Token Using Personal API Token

> To fetch app versions and download the binary, we first need to obtain an access token using a Personal API Token (PAT).

Reviewer: ```suggestion
To fetch app versions and download the binary, you first need to obtain an access token using a Personal API Token (PAT).
```
